### PR TITLE
refactor: rewrite JunitParser for junitparser > 2

### PR DESCRIPTION
**Pull request by arys-splunk**
**Background**

PSA uses `junitparser` library (https://github.com/weiwei/junitparser) to parse results after tests in XML format.
PSA was using version 1.6.3, but `junitparser` 2.0.0 has a breaking change (weiwei/junitparser@e1270eb), now `testcase.result` is a list, instead of being a single element for `junitparser` < 2.

**What was done**

1. Refactored `JunitParser` , so it accepts new testcases from `junitparser` > 2.
2. Moved docker-based test for JunitParser to unit tests.
2. Refactored unit tests for `JunitParser`

### CODE REVIEW

File: tests/unit/tests_standard_lib/test_utilities/test_junit_parser.py [hunk 1]
```diff
@@ -1,264 +1,72 @@
...
+            "fields": "field_4",
+            "data_set": "dataset_4",
+            "fields_type": "required",
+            "tag_stanza": 'dataset_name="dataset_4"',
+            "status": "skipped",
+            "test_property": "-",
+        },
     ]
+    jp = JunitParser(
+        os.path.join(os.path.dirname(__file__), "test_data", "sample_junit.xml")
```
Comment (okashaev-splunk) [inline tests/unit/tests_standard_lib/test_utilities/test_junit_parser.py:58]: It looks like the test should load .../test_utilities/test_data/sample_junit.xml, parse it and compare with expected result, correct? I see above that sample_junit.xml is empty so test should probably fail.

**Review by okashaev-splunk — CHANGES_REQUESTED** [review]

File: tests/unit/tests_standard_lib/test_utilities/test_junit_parser.py (hunk 1 above)
Comment (arys-splunk) [reply tests/unit/tests_standard_lib/test_utilities/test_junit_parser.py:58]: Github shows it as "File renamed without changes.", I just moved that file to another directory.

Comment (okashaev-splunk) [reply tests/unit/tests_standard_lib/test_utilities/test_junit_parser.py:58]: Yes, sorry, your are right, I misunderstood the 0 value

**Review by okashaev-splunk — APPROVED** [review]

**Review by hsekowski-splunk — APPROVED** [review]

**Pull request merged**